src/tree.c

```diff
@@ -281,18 +281,24 @@ const git_oid *git_tree_entry_id(const git_tree_entry *entry)
 	return entry->oid;
 }
 
-git_otype git_tree_entry_type(const git_tree_entry *entry)
+static git_otype filemode2type(git_filemode_t filemode)
```

Comment:
We have `git_object__type_from_filemode` which we may be able to use.  (I didn't think to spelunk for other users when I added it, oops.)